### ISSUE TITLE
System provider fails on OS X

### ISSUE THREAD

**Issue by jacobjennings** [poise/poise-python#15]
```
Recipe: poise-python::default
  * python_runtime[2] action install[2015-11-16T08:40:42-08:00] INFO: Processing python_runtime[2] action install (poise-python::default line 20)

    * poise_languages_system[python] action install[2015-11-16T08:40:42-08:00] INFO: Processing poise_languages_system[python] action install (/Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/mixin.rb line 32)


      ================================================================================
      Error executing action `install` on resource 'poise_languages_system[python]'
      ================================================================================

      TypeError
      ---------
      no implicit conversion of Array into String

      Cookbook Trace:
      ---------------
      /Users/jacob/sprout-wrap/cookbooks/homebrew/libraries/homebrew_package.rb:97:in `get_version_from_formula'
      /Users/jacob/sprout-wrap/cookbooks/homebrew/libraries/homebrew_package.rb:74:in `current_installed_version'
      /Users/jacob/sprout-wrap/cookbooks/homebrew/libraries/homebrew_package.rb:44:in `load_current_resource'
      /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:189:in `block (2 levels) in patch_load_current_resource!'
      /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:172:in `block in run_package_action'
      /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:162:in `each'
      /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:162:in `run_package_action'
      /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:100:in `action_install'
      /Users/jacob/sprout-wrap/cookbooks/poise/files/halite_gem/poise/helpers/notifying_block.rb:69:in `notifying_block'
      /Users/jacob/sprout-wrap/cookbooks/poise-python/files/halite_gem/poise_python/python_providers/base.rb:44:in `action_install'

      Resource Declaration:
      ---------------------
      # In /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/mixin.rb

       32:         poise_languages_system options['package_name'] || system_package_name do
       33:           # Otherwise use the default install action.
       34:           action(:upgrade) if options['package_upgrade']
       35:           parent new_resource
       36:           # Don't pass true because we want the default computed behavior for that.
       37:           dev_package options['dev_package'] unless options['dev_package'] == true
       38:           dev_package_overrides dev_package_overrides
       39:           package_version options['package_version'] if options['package_version']
       40:           version options['version']
       41:         end
       42:       end

      Compiled Resource:
      ------------------
      # Declared in /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/mixin.rb:32:in `install_system_packages'

      poise_languages_system("python") do
        action [:install]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        declared_type :poise_languages_system
        cookbook_name :"poise-python"
        parent # Declared in /Users/jacob/sprout-wrap/cookbooks/poise-python/recipes/default.rb:20:in `from_file'

      python_runtime("2") do
        action [:install]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        declared_type :python_runtime
        cookbook_name :"poise-python"
        recipe_name "default"
        pip_version true
        setuptools_version true
        version "2"
        virtualenv_version true
        wheel_version true
      end

        version "2"
        package_name "python"
      end

[2015-11-16T08:40:42-08:00] INFO: Running queued delayed notifications before re-raising exception

    ================================================================================
    Error executing action `install` on resource 'python_runtime[2]'
    ================================================================================

    TypeError
    ---------
    poise_languages_system[python] (/Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/mixin.rb line 32) had an error: TypeError: no implicit conversion of Array into String

    Cookbook Trace:
    ---------------
    /Users/jacob/sprout-wrap/cookbooks/homebrew/libraries/homebrew_package.rb:97:in `get_version_from_formula'
    /Users/jacob/sprout-wrap/cookbooks/homebrew/libraries/homebrew_package.rb:74:in `current_installed_version'
    /Users/jacob/sprout-wrap/cookbooks/homebrew/libraries/homebrew_package.rb:44:in `load_current_resource'
    /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:189:in `block (2 levels) in patch_load_current_resource!'
    /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:172:in `block in run_package_action'
    /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:162:in `each'
    /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:162:in `run_package_action'
    /Users/jacob/sprout-wrap/cookbooks/poise-languages/files/halite_gem/poise_languages/system/resource.rb:100:in `action_install'
    /Users/jacob/sprout-wrap/cookbooks/poise/files/halite_gem/poise/helpers/notifying_block.rb:69:in `notifying_block'
    /Users/jacob/sprout-wrap/cookbooks/poise-python/files/halite_gem/poise_python/python_providers/base.rb:44:in `action_install'

    Resource Declaration:
    ---------------------
    # In /Users/jacob/sprout-wrap/cookbooks/poise-python/recipes/default.rb

     20: python_runtime '2' if node['poise-python']['install_python2']

    Compiled Resource:
    ------------------
    # Declared in /Users/jacob/sprout-wrap/cookbooks/poise-python/recipes/default.rb:20:in `from_file'

    python_runtime("2") do
      action [:install]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :python_runtime
      cookbook_name :"poise-python"
      recipe_name "default"
      pip_version true
      setuptools_version true
      version "2"
      virtualenv_version true
      wheel_version true
    end

[2015-11-16T08:40:42-08:00] INFO: Running queued delayed notifications before re-raising exception
```

**Comment by coderanger** [comment]
Guessing you are trying to use this on OS X? I've not tested it and it sounds like the Homebrew package provider doesn't support multi-package processing currently.

**Comment by coderanger** [comment]
Fixed in `poise-languages` 1.3.1. Should fall back to non-multi-install now.
